doc: kconfig: Cover a common gotcha related to "stuck" symbol values

ulfalizer opened the pull request:
Commits:

```
Explain a common subtle Kconfig gotcha where it might appear that
symbols with prompts are incorrectly "stuck" to a particular value in
the menuconfig interface. I've seen it come up several times now, e.g.
in the discussion that lead to
https://github.com/zephyrproject-rtos/zephyr/pull/20702. I also suspect
it's what's causing
https://github.com/zephyrproject-rtos/zephyr/issues/20673.

Also give suggestions for how to make things behave as intended.
```

```
doc: kconfig: Mention guiconfig on tips page
    
Mention guiconfig as well wherever menuconfig is mentioned.
```

Also updated the ci-tools repo to pass CI:

```
west.yml: Update ci-tools to whitelist STACK_SIZE

Get this commit in, needed to pass CI with the doc changes in this PR:

    check_compliance.py: Add STACK_SIZE to Kconfig symbol whitelist

    Needed for https://github.com/zephyrproject-rtos/zephyr/pull/20722
```

aescolar reviewed (approved):
It's always nice to have more information about the workings of kconfig :)

carlescufi commented on doc/guides/kconfig/index.rst:
```diff
@@ -326,6 +326,97 @@ error-prone, since it can be hard to spot that the same dependency is added
 twice.
 
 
+Why aren't other symbols getting updated when I change a symbol's value?
```
Shouldn't this add "when I use menuconfig or guiconfig"?

dbkinder replied:
Maybe instead: ``Getting a dependent symbol value to update when using menuconfig``

ulfalizer replied:
@carlescufi 
Felt like it made it a bit long.

@dbkinder 
Will work normally, e.g. when using `depends on`. It's only in this special case that it breaks.

ulfalizer replied:
Changed the heading to `"Stuck" symbols in the menuconfig`. Is that better?

Could make it "menuconfig or guiconfig" (and do the same in other places that mention menuconfig), but I think it might be a bit redundant.

ulfalizer replied:
@carlescufi ping

carlescufi replied:
I'd really prefer my original proposal. I think this is common enough that it warrants being very explicit about the actual problem description.

ulfalizer replied:
Added another commit to the PR that mentions `guiconfig` wherever `menuconfig` is mentioned.

dbkinder reviewed (approved):
LGTM, thanks

ulfalizer commented:
Needs https://github.com/zephyrproject-rtos/ci-tools/pull/109.

mbolivar reviewed (approved):
Thanks! This is quite clear.

joerchan commented:
@ulfalizer The dependant PR you mentioned is merged. Would rebasing this PR fix the Kconfig failure?

ulfalizer committed:
doc: kconfig: Cover a common gotcha related to "stuck" symbol values

Explain a common subtle Kconfig gotcha where it might appear that
symbols with prompts are incorrectly "stuck" to a particular value in
the menuconfig interface. I've seen it come up several times now, e.g.
in the discussion that lead to
https://github.com/zephyrproject-rtos/zephyr/pull/20702. I also suspect
it's what's causing
https://github.com/zephyrproject-rtos/zephyr/issues/20673.

Also give suggestions for how to make things behave as intended.

Signed-off-by: Ulf Magnusson <Ulf.Magnusson@nordicsemi.no>

ulfalizer committed:
doc: kconfig: Mention guiconfig on tips page

Mention guiconfig as well wherever menuconfig is mentioned.

Signed-off-by: Ulf Magnusson <Ulf.Magnusson@nordicsemi.no>

ulfalizer commented:
> @ulfalizer The dependant PR you mentioned is merged. Would rebasing this PR fix the Kconfig failure?

Also needs an update of ci-tools in `west.yml`. I'll add it.

ulfalizer committed:
west.yml: Update ci-tools to whitelist STACK_SIZE

Get this commit in, needed to pass CI with the doc changes in this PR:

    check_compliance.py: Add STACK_SIZE to Kconfig symbol whitelist

    Needed for https://github.com/zephyrproject-rtos/zephyr/pull/20722.

Signed-off-by: Ulf Magnusson <Ulf.Magnusson@nordicsemi.no>

ulfalizer commented:
PR rebased now. Also added the required ci-tools update, and added another commit to mention `guiconfig` wherever `menuconfig` is mentioned on the Kconfig tips page.
